text/XXXX-replace_publish_clone_with_push_pull.md

```diff
@@ -0,0 +1,160 @@
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+## New Commands
+I'm going to focus on writing help text that explains the proposed commands. The help text should describe the commands well enough to understand what this RFC is proposing.
+
+
+### push: upload datasets to qri peers
+```
+$ qri push --help
+Push sends datasets to a remote qri node. A push updates the dataset log on the
+remote and sends as many versions as the remote is willing to accept. A remote
+peer determines it's own rules on which versions to accept, but by default
+favour the latest versions of a dataset.
+
+If no remote is specified, qri pushes to the registry. But pushes 
+can just as easily send to another peer that is configured to accept datasets 
+from you. For details on  setting peers for dataset exchange read this doc:
+https://qri.io/docs/hosting-datasets
+
+Usage:
+  qri push [REMOTE] DATASET [DATASET] [flags]
+
+Aliases:
+  publish
+
+Examples:
+  # push a dataset to the registry
+  $ qri push me/dataset
+
+  # push a specific version of a dataset to the registry:
+  $ qri push me/dataset@/ipfs/QmHashOfVersion
+
+  # push only the latest two versions to a peer named xiao
+  $ qri push xiao me/dataset --revisions 2
+
+Flags:
+  -l, --logs               send only dataset history
+  -r, --revisions          send only the lastest commit data
+```
+
+### pull: download datasets from the qri network
+```
+$ qri pull --help
+Pull downloads datasets and stores them locally, fetching the dataset log and
+dataset version(s). By default pull only fetches the latest version of a dataset
+
+By default pull fetches 
+
+Because pull grabs the latest version by default, it's common to 
+
+If qri connect is running
+
+Usage:
+  qri pull [REMOTE] [DATASET] [flags]
+
+Examples:
+  # download a dataset log and latest version
+  $ qri pull b5/world_bank_population
+
+  # download a dataset log and all versions
+  $ qri pull b5/world_bank_population --all
+
+  # pull a dataset from 
+  $ qri pull ramfox b5/world_bank_population
+
+Flags:
+  -l, --logs              save only dataset history
+  -r, --revisions         fetch a set number of versions from the latest commit
+  -a, --all               fetch all versions, overrides revisions
+  -k, --keep-files        when in a working directory, don't update files
+
+```
+
+`revisions` and `all` flags mimic the way `qri remove` works.
+
+### clone: pull and checkout a dataset
+
+We already have "clone" language in qri cloud, and have intended to rename `add` to `clone` for some time. The problem is add isn't the same thing as clone when compared to git. Removing the idea of publish & unpublish gives us back a little cognitive overhead that we can use for the classic combo of "get the latest version and checkout a directory
```

Comment:
This made me realize I didn’t see anything in the above about unpublishing. I’m assuming it’ll be part of `qri push`, but don’t feel clear on how it would work.

It doesn’t *feel* like there’s an obvious analogue to Git, where the full command is more like “push this local pointer to a tree of hashes to this remote pointer to a tree of hashes” — and you unpublish by pushing nothing to a previously existing branch pointer. (And, arguably, this is among the more confusing parts of Git that people use regularly.)

Reply:
+1, we need clarification on unpublishing.  If I pushed directly to a peer, presumably I can't "take it back", but I *can* prevent that peer from seeing updates to the dataset log or pulling new versions.

If I push to cloud, does cloud check my peer for new updates, or just think it has the latest until I push again?  If I push to a peer, can the peer check for updates on its own or do I have to push again?

Reply:
point taken on the unpublishing bit missing. I'ver written another rfc that proposes `qri drop` for this: https://github.com/qri-io/rfcs/pull/53